### PR TITLE
AESinkAudioTrack: Help broken firmwares to make kodi ignore broken delay

### PR DESCRIPTION
**Pull request by fritsch**
Kodi can supervise the sink for being stuck, which means: if you open a sink with 160 ms buffer and it does not move while eating more than twice of the buffer's audio data, it is considered a bug. Sadly - there is a whole lot of broken firmwares out there, one of the leards: Ugoos. Which is the reason I make this opt-in. That means that FireTV Cube 3rd Gen and others would need to enable this - once again - via an advanced settings.

```
<advancedsettings>
<audio>
<superviseaudiodelay>true</superviseaudiodelay> 
</audio>
</advancedsettings>
```

See forum report: https://forum.kodi.tv/showthread.php?tid=376090 - I briefly informed in Slack some weeks ago and just waited to see more broken devices. Yeah, seen now the Ugoos twice ...

For now it's only the Ugoos, while I did not get other reports. The alternative to this change would be a "negative list" .... which we don't want to maintain. Sad thing is, that FireTV Cube 3rd Gen and the other boxes that need this are a bigger number.

Open for suggestions.

### CODE REVIEW

File: xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp [hunk 1]
```diff
@@ -825,7 +827,7 @@ unsigned int CAESinkAUDIOTRACK::AddPackets(uint8_t **data, unsigned int frames,
     const double max_stuck_delay_ms = m_audiotrackbuffer_sec_orig * 2000.0;
     const double stime_ms = 1000.0 * frames / m_format.m_sampleRate;
 
-    if (m_stuckCounter * stime_ms > max_stuck_delay_ms)
+    if (!m_ignoreBrokenAudioDelay && (m_stuckCounter * stime_ms > max_stuck_delay_ms))
```
Comment (thexai) [inline xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp:830]: I find this a bit unintuitive as default advancedsetting is enabled and here negated and in `AESinkAUDIOTRACK.h` variable is initialized to false.

Maybe reword to "m_trackAudioDelay" or similar and default = false on all places (if the intention is not track by default).

Comment (fritsch) [reply xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp:830]: Infact I did not think too much about it ... I hate such workarounds ... ;-) but your suggestion is good.

**Comment by fritsch** [comment]
Let's keep that just here ... want to see if it's really needed or could be inversed ... after next beta lots of additional testers will appear ...

File: xbmc/settings/AdvancedSettings.h [hunk 2]
```diff
@@ -154,7 +154,8 @@ class CAdvancedSettings : public ISettingCallback, public ISettingsHandler
     float m_videoIgnorePercentAtEnd;
     float m_audioApplyDrc;
     unsigned int m_maxPassthroughOffSyncDuration = 10; // when 10 ms off adjust
-    bool m_AllowMultiChannelFloat = false; // Android only switch to be remved in v22
+    bool m_AllowMultiChannelFloat = false; // Android only switch to be removed in v22
+    bool m_superviseAudioDelay = false; // Android only to not correct broken audio firmwares
```
Comment (thexai) [inline xbmc/settings/AdvancedSettings.h:158]:
```suggestion
    bool m_superviseAudioDelay = false; // Android only to correct broken audio firmwares
```

or maybe:
// Android only workaround to correct broken audio firmwares

since both code to supervise delay and setting to disable it are workarounds

**Review by thexai — APPROVED** [review]
Approved conditioned to improve code comment a bit

**Comment by fritsch** [comment]
jenkins build this please
